sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -317,4 +318,41 @@ case class ShuffledHashJoinExec(
       v => s"$v = $thisPlan.buildHashedRelation(inputs[1]);", forceInline = true)
     HashedRelationInfo(relationTerm, keyIsUnique = false, isEmpty = false)
   }
+
+  /**
+   * Generates the code for anti join.
+   * Handles NULL-aware anti join (NAAJ) separately here.
+   */
+  protected override def codegenAnti(ctx: CodegenContext, input: Seq[ExprCode]): String = {
```

Comment:
given we refactored non-code-gen `nullAwareAntiJoin` to `HashJoin`, I think we can also refactor code-gen `codegenNullAwareAntiJoin` from `BroadcastHashJoinExec` and `ShuffledHashJoinExec` to `HashJoin` to avoid code duplication.